src/Core/tests/DeviceTests/Stubs/PickerStub.cs

```diff
@@ -12,7 +13,24 @@ public partial class PickerStub : StubBase, IPicker
 
 		public IList<string> Items { get; set; } = new List<string>();
 
-		public IList ItemsSource { get; set; }
+		IList _itemsSource;
+
+		public IList ItemsSource
+		{
+			get => _itemsSource;
+			set => SetProperty(ref _itemsSource, value, onChanged: OnItemsSourceChanged);
+		}
+
+		void OnItemsSourceChanged(IList oldValue, IList newValue)
+		{
+			if (ItemsSource == null)
+				return;
+
+			Items.Clear();
```

Comment:
I think you have to clear the list before check ItemsSource is null

Reply:
Look @mohachouch  our WPF guru :D 

This is just a test so i _think_ is fine. 

For when a WPF backend to MAUI @mohachouch  ? i can help :P